qcodes/instrument_drivers/american_magnetics/AMI430.py

```diff
@@ -2,15 +2,97 @@
 import logging
 import time
 from functools import partial
+from warnings import warn
 
 import numpy as np
 
-from qcodes import Instrument, IPInstrument
+from qcodes import Instrument, IPInstrument, InstrumentChannel
 from qcodes.math.field_vector import FieldVector
-from qcodes.utils.validators import Numbers, Anything
+from qcodes.utils.validators import Bool, Numbers, Ints, Anything
 
 log = logging.getLogger(__name__)
 
+class AMI430Exception(Exception):
```

Comment:
I do not quite understand why we need to make a custom exception. Can you explain?

Reply:
Just so that errors that are specific to the instrument state are differentiated from user errors (like `ValueError`s) or python errors. I suppose they can be replaced by `RuntimeError`s, but this seems like a poor fit given that they are used everywhere, and the error type suggests the error occured within python.

In particular, I imagined using it to detect a failure to sweep field in my own code, then automatically taking action.

Reply:
Yeah I see your point.